pkg/cluster/managed.go

```diff
@@ -165,3 +171,26 @@ func (c *Cluster) setupEtcdProxy(ctx context.Context, etcdProxy etcd.Proxy) {
 		}
 	}()
 }
+
+// deleteNodePasswdSecret wipes out the node password secret after restoration
+func (c *Cluster) deleteNodePasswdSecret(ctx context.Context) {
+	t := time.NewTicker(5 * time.Second)
+	defer t.Stop()
+	for range t.C {
+		nodeName := os.Getenv("NODE_NAME")
+		if nodeName == "" {
+			logrus.Infof("waiting for node name to be set")
+			continue
+		}
+		secretsClient := c.runtime.Core.Core().V1().Secret()
+		if err := nodepassword.Delete(secretsClient, nodeName); err != nil {
+			if apierrors.IsNotFound(err) {
+				return
+			}
+			logrus.Warnf("failed to delete old node password secret: %v", err)
```

Comment:
Do we want any retry logic here or account for another server trying to do the same thing?

Reply:
so the retry logic will attempt to delete, if another server is doing the delete it will be modifying the object and this one will fail and retry in 5 seconds, on the next retry the secret will be deleted and it will return nil

Reply:
I guess I was thinking more along the lines of the clientgo retry logic. If this will suffice, given it's a "delete" operation, I'm cool with that. :)